src/python/pants/option/options.py

```diff
@@ -389,35 +389,35 @@ def for_scope(
         :raises pants.option.errors.ConfigValidationError: if the scope is unknown.
         """
 
-        values_builder = OptionValueContainerBuilder()
-        flags_in_scope = self._scope_to_flags.get(scope, [])
-        parse_args_request = self._make_parse_args_request(flags_in_scope, values_builder)
-        legacy_values = self.get_parser(scope).parse_args(
-            parse_args_request, log_warnings=log_parser_warnings
-        )
+        native_values = self.get_parser(scope).parse_args_native(self._native_parser)
```

Comment:
Now we always create the native values but only create the legacy values if we need to do the discrepancy check, which is the converse of what we did before.